Source/CombatExtended/CombatExtended/Projectiles/BulletCE.cs

```diff
@@ -35,28 +40,21 @@ protected override void Impact(Thing hitThing)
             Map map = base.Map;
             LogEntry_DamageResult logEntry = null;
 
-            if (logMisses
-                ||
-                (!logMisses
-                    && hitThing != null
-                    && (hitThing is Pawn
-                        || hitThing is Building_Turret)
-                 ))
+            if (cookOff == false && (logMisses || hitThing is Pawn || hitThing is Building_Turret))
```

Comment:
`== false` is never a good thing in C#.  You're counting on the compiler or the JIT fixing it.  It should be `!cookOff`.

Reply:
I doubt the compiler would miss this given how obvious it is, So I prioritised the readability; especially given how cryptic the original statement looked. I am happy to change it if you prefer the exclamation point negation.

Reply:
To a very novice programmer, `== false` might be more readable, but it really isn't.  Least surprise and all that.  Same thing with line 32.  

I did verify that `csc` is smart enough to treat them the same, and discovered that `mcs` is much smarter than `csc`.  Kinda a pity that `mcs` lags behind feature-wise so badly, as it produces significantly better code.

Regardless, no place else do we use `== false` or `== true`, and I haven't encountered a significant C# codebase that does, so we should aim for consistency.